src/main/java/org/ibp/api/brapi/v2/observation/ObservationUnitResourceBrapi.java

```diff
@@ -0,0 +1,116 @@
+package org.ibp.api.brapi.v2.observation;
+
+import com.fasterxml.jackson.annotation.JsonView;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.generationcp.middleware.domain.search_request.brapi.v2.ObservationUnitsSearchRequestDto;
+import org.generationcp.middleware.exceptions.MiddlewareException;
+import org.generationcp.middleware.manager.api.SearchRequestService;
+import org.generationcp.middleware.service.api.phenotype.PhenotypeSearchDTO;
+import org.generationcp.middleware.service.api.phenotype.PhenotypeSearchRequestDTO;
+import org.ibp.api.brapi.v1.common.BrapiPagedResult;
+import org.ibp.api.brapi.v1.common.EntityListResponse;
+import org.ibp.api.brapi.v1.common.Metadata;
+import org.ibp.api.brapi.v1.common.Pagination;
+import org.ibp.api.brapi.v1.common.Result;
+import org.ibp.api.domain.common.PagedResult;
+import org.ibp.api.java.dataset.DatasetService;
+import org.ibp.api.java.study.StudyService;
+import org.ibp.api.rest.common.PaginatedSearch;
+import org.ibp.api.rest.common.SearchSpec;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Api(value = "BrAPI Observation Unit Services")
+@Controller
+public class ObservationUnitResourceBrapi {
+
+	@Autowired
+	private DatasetService datasetService;
+
+	@Autowired
+	private SearchRequestService searchRequestService;
+
+	@Autowired
+	private StudyService studyService;
+
+	@ApiOperation(value = "Get Observation Unit search", notes = "Get the results of a Observation Unit search request")
+	@RequestMapping(value = "/{crop}/brapi/v2/search/observationunits/{searchResultsDbid}", method = RequestMethod.GET)
+	@ResponseBody
+	@JsonView(PhenotypeSearchDTO.View.PhenotypeBrapiV2.class)
+	public ResponseEntity<EntityListResponse<PhenotypeSearchDTO>> getSearchObservationUnit(
```

Comment:
@vmaletta rename to getObservationUnitsSearch

Reply:
ok